### PR TITLE
[SAPBTPCFS-7876] Optimize handling of non-transient errors

### PR DESCRIPTION
**Pull request by I065450**
https://github.com/SAP/sap-btp-service-operator/issues/353
add an option to ignore a non transient error for a period of time for cases when there is a dependency between two resources.

### CODE REVIEW

File: api/utils/utils.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+)
+
+func ValidateNonTransientTimestampAnnotation(log logr.Logger, resource api.SAPBTPResource) error {
+
+	sinceAnnotation, exist, err := GetTimeSinceIgnoreNonTransientAnnotationTimestamp(log, resource)
+	if err != nil {
+		return err
+	}
+	if exist && sinceAnnotation < 0 {
+		return fmt.Errorf("annotation %s cannot be a future timestamp", api.IgnoreNonTransientErrorTimestampAnnotation)
```
Comment (I065450) [inline api/utils/utils.go:18]: @danielforsap review error message

File: api/utils/utils.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+func GetTimeSinceIgnoreNonTransientAnnotationTimestamp(log logr.Logger, resource api.SAPBTPResource) (time.Duration, bool, error) {
+	annotation := resource.GetAnnotations()
+	if annotation != nil {
+		if _, ok := annotation[api.IgnoreNonTransientErrorAnnotation]; ok {
+			log.Info("ignoreNonTransientErrorAnnotation annotation exist- checking timeout")
+			annotationTime, err := time.Parse(time.RFC3339, annotation[api.IgnoreNonTransientErrorTimestampAnnotation])
+			if err != nil {
+				log.Error(err, fmt.Sprintf("failed to parse %s", api.IgnoreNonTransientErrorTimestampAnnotation))
+				return time.Since(time.Now()), false, fmt.Errorf("annotation %s is not a valid timestamp", api.IgnoreNonTransientErrorTimestampAnnotation)
```
Comment (I065450) [inline api/utils/utils.go:46]: @danielforsap review error message

File: api/v1/servicebinding_validating_webhook.go [hunk 3]
```diff
@@ -124,3 +130,12 @@ func (sb *ServiceBinding) validateCredRotatingConfig() error {
 
 	return nil
 }
+
+func (sb *ServiceBinding) validateAnnotations() error {
+	if sb.Annotations != nil {
+		if _, ok := sb.Annotations[api.IgnoreNonTransientErrorAnnotation]; ok {
+			return fmt.Errorf("annotation %s is not suppoted in service binding", api.IgnoreNonTransientErrorAnnotation)
```
Comment (I065450) [inline api/v1/servicebinding_validating_webhook.go:137]: @danielforsap review error message

File: api/utils/utils.go (hunk 2 above)
Comment (danielforsap) [reply api/utils/utils.go:46]: @I065450 can we list the timestamps that are valid? Is it about their format or the actual time. Or both?

File: api/utils/utils.go (hunk 1 above)
Comment (danielforsap) [reply api/utils/utils.go:18]: @I065450  Pls give me a concrete example.
I am trying to understand the context.
Also, what comes before and after these messages? Are they wrapped in larger text or standalone sentences?
Thanks

File: api/v1/servicebinding_validating_webhook.go (hunk 3 above)
Comment (danielforsap) [reply api/v1/servicebinding_validating_webhook.go:137]: The specified annotation '%s' is not supported within the service binding object. 
@I065450  can we say that?

File: api/utils/utils.go (hunk 1 above)
Comment (danielforsap) [reply api/utils/utils.go:18]: @I065450 Annotation cannot be a date in the future.

File: api/utils/utils.go (hunk 2 above)
Comment (I065450) [reply api/utils/utils.go]: the format is time.RFC3339

**Review by kerenlahav — APPROVED** [review]

**Pull request merged**